.github/workflows/pr.ci.yml

```diff
@@ -0,0 +1,114 @@
+# This workflow will do a clean install of the dependencies and run tests across different versions
+#
+# Requires scripts:
+# - bin/pr-check.sh
+# - bin/pr.sh
+
+name: elixir / pr
+
+on: pull_request
+
+jobs:
+  precheck:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout PR
+        uses: actions/checkout@v2
+
+      - name: Use Elixir
+        uses: actions/setup-elixir@v1
+        with:
+          otp-version: '22.2'
+          elixir-version: '1.10.0'
+
+      - name: Retrieve Mix Dependencies Cache
+        uses: actions/cache@v1
+        id: mix-cache # id to use in retrieve action
+        with:
+          path: deps
+          key: ${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-mix-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
+
+      - name: Install Mix Dependencies
+        if: steps.mix-cache.outputs.cache-hit != 'true'
+        run: |
+          mix local.rebar --force
+          mix local.hex --force
+          mix deps.get
+
+      - name: Build Project
+        run: mix
+
+      - name: Find Trailing Whitespace
+        uses: harupy/find-trailing-whitespace@master
```

Comment:
This is a third party action, so you should pin it to a commit hash rather than `@master` or a tag for security reasons: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions

It's somewhat tedious to keep commit hashes up-to-date, so I'd personalyl remove this third party action and find another solution to deal with trailing whitespace or not bother with it at all.

TL;DR: If that actions is compromised, a malicious actor could gain write access to the repo through the GITHUB_TOKEN.

Reply:
Thanks. I moved the whitespace checking into the bash file. Using the same mechanism we previously had in Travis.